cmd/destroy/destroy.go

```diff
@@ -218,345 +200,49 @@ func Destroy(ctx context.Context) *cobra.Command {
-	oktetoLog.SetStage(fmt.Sprintf("Destroying by label '%s'", deployedBySelector))
-	if err := dc.nsDestroyer.DestroyWithLabel(ctx, opts.Namespace, deleteOpts); err != nil {
-		oktetoLog.Infof("could not delete all the resources: %s", err)
-		if err := dc.configMapHandler.setErrorStatus(ctx, cfg, data, err); err != nil {
-			return err
-		}
-		return err
-	}
-
-	oktetoLog.SetStage("Destroying configmap")
-
-	if err := dc.configMapHandler.destroyConfigMap(ctx, cfg, namespace); err != nil {
-		return err
-	}
-
-	return commandErr
-}
-
-func (dc *destroyCommand) destroyHelmReleasesIfPresent(ctx context.Context, opts *Options, labelSelector string) error {
-	sList, err := dc.secrets.List(ctx, opts.Namespace, labelSelector)
-	if err != nil {
-		return err
-	}
-
-	oktetoLog.Debugf("checking if application installed something with helm")
-	helmReleases := map[string]bool{}
-	for _, s := range sList {
-		if s.Type == model.HelmSecretType && s.Labels[ownerLabel] == helmOwner {
-			helmReleaseName, ok := s.Labels[nameLabel]
-			if !ok {
-				continue
-			}
-
-			helmReleases[helmReleaseName] = true
-		}
-	}
-
-	// If the application to be destroyed was deployed with helm, we try to uninstall it to avoid to leave orphan release resources
-	for releaseName := range helmReleases {
-		oktetoLog.Debugf("uninstalling helm release '%s'", releaseName)
-		cmd := fmt.Sprintf(helmUninstallCommand, releaseName)
-		cmdInfo := model.DeployCommand{Command: cmd, Name: cmd}
-		oktetoLog.Information("Running '%s'", cmdInfo.Name)
-		if err := dc.executor.Execute(cmdInfo, opts.Variables); err != nil {
-			oktetoLog.Infof("could not uninstall helm release '%s': %s", releaseName, err)
-			if !opts.ForceDestroy {
-				return err
-			}
-		}
-	}
-
-	return nil
-}
-
 func getTempKubeConfigFile(name string) string {
 	tempKubeconfigFileName := fmt.Sprintf("kubeconfig-destroy-%s-%d", name, time.Now().UnixMilli())
 	return filepath.Join(config.GetOktetoHome(), tempKubeconfigFileName)
 }
 
-func (dc *destroyCommand) runDestroyAll(ctx context.Context, opts *Options) error {
-	oktetoLog.Spinner(fmt.Sprintf("Deleting all in %s namespace", opts.Namespace))
-	oktetoLog.StartSpinner()
-	defer oktetoLog.StopSpinner()
-
-	if err := dc.oktetoClient.Namespaces().DestroyAll(ctx, opts.Namespace, opts.DestroyVolumes); err != nil {
-		return err
-	}
-
-	waitCtx, ctxCancel := context.WithCancel(ctx)
-	defer ctxCancel()
-
-	stop := make(chan os.Signal, 1)
-	defer close(stop)
-
-	signal.Notify(stop, os.Interrupt)
-	exit := make(chan error, 1)
-	defer close(exit)
-
-	var wg sync.WaitGroup
-
-	wg.Add(1)
-	go func(wg *sync.WaitGroup) {
-		defer wg.Done()
-		exit <- dc.waitForNamespaceDestroyAllToComplete(waitCtx, opts.Namespace)
-	}(&wg)
+func getDestroyer(k8sClient *kubernetes.Clientset, okK8sClient okteto.K8sClientProvider, nsDestroyer destroyer, executor executor.ManifestExecutor, okClient *okteto.OktetoClient, opts *Options) (destroyInterface, error) {
+	var (
```

Comment:
I think it's easier to read if we write it as a one-line var
```suggestion
	var deployer destroyInterface
```